transcript/src/basic.rs

```diff
@@ -5,64 +5,45 @@ use poseidon::poseidon_permutation::PoseidonPermutation;
 
 use crate::Challenge;
 
-#[derive(Clone)]
-pub struct Transcript<E: ExtensionField> {
-    permutation: PoseidonPermutation<E::BaseField>,
-}
-
-impl<E: ExtensionField> Transcript<E> {
-    /// Create a new IOP transcript.
-    pub fn new(label: &'static [u8]) -> Self {
-        let mut perm = PoseidonPermutation::new(core::iter::repeat(E::BaseField::ZERO));
-        let label_f = E::BaseField::bytes_to_field_elements(label);
-        perm.set_from_slice(label_f.as_slice(), 0);
-        perm.permute();
-        Self { permutation: perm }
+/// The Transcript trait
```

Comment:
Maybe move the trait to `lib.rs` or a separate file like `common.rs`?

Reply:
Done. Traits have been moved into `lib.rs`